helpers/splitWithSearchTerm.tsx

```diff
@@ -4,30 +4,14 @@
 		Inputs: searchTerm='bon', str = 'bonJourbon'
 		Output: [bon, Jour, bon]
 */
-export const splitWithSearchTerm = (str: string, searchTerm: string) => {
-  const lowerCaseSearchTerm = searchTerm.toLowerCase()
-  const splitArr = str.toLowerCase().split(lowerCaseSearchTerm)
-  // tracker is used for `str.substr` to extract characters from the original string at the correct places
-  let tracker = 0
-
-  const res = splitArr.reduce(
-    (acc: string[], word: string, splitArrIndex: number) => {
-      // converts words back into their original capitalization
-      const originalWord = str.substr(tracker, word.length)
-      acc.push(originalWord)
-      tracker += word.length
-
-      // used to prevent over-adding of searchTerm back into array
-      if (splitArrIndex === splitArr.length - 1) return acc
 
-      // convert searchTerm to match original capitalization of string
-      const correctSearchCapitalization = str.substr(tracker, searchTerm.length)
-      acc.push(correctSearchCapitalization)
-      tracker += searchTerm.length
-      return acc
-    },
-    []
-  )
-
-  return res
+export const splitWithSearchTerm = (str: string, searchTerm: string) => {
+  /*
+	Using regex with split will return an array with the separator included.
+	The `i` flag is the case insensitive flag, used to ensure that the returned array
+	will include the searchTerm, regardless of the original capialization of the searchTerm
```

Comment:
Just a correction in the documentation. What actually makes `searchTerm` be included in the returned array are the parenthesis. The `i` flag is only for case insensitive match.

As per the MDN example:

> If `separator` is a regular expression that contains capturing parentheses `( )`, matched results are included in the array.

Reply:
Ah I see there has been a misunderstanding, because I indeed was trying to explain in my comment that the `i` flag is used for case sensitivity as you have thought I didn't. 

Thank you for bringing this to my attention, I have updated my comment in hopes that people are less confused by my `i` flag explanation.